src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/BulkPrintService.java

```diff
@@ -86,13 +90,29 @@ private UUID printDocuments(CaseDetails caseDetails, BulkPrintDocument coverShee
 
     private BulkPrintDocument generateApplicantCoverSheet(CaseDetails caseDetails, String authorisationToken) {
         CaseDocument applicantCoverSheet = coverSheetService.generateApplicantCoverSheet(caseDetails, authorisationToken);
-        caseDetails.getData().put(BULK_PRINT_COVER_SHEET_APP, applicantCoverSheet);
+
+        if (isApplicantAddressConfidential(caseDetails.getData())) {
+            log.info("Case {}, has been marked as confidential. Adding coversheet to confidential field", caseDetails.getId());
```

Comment:
Not a blocker but might be nice to be more verbose with these and specify in the log message that its for the applicant confidential field etc, and same in the resp method below